skcosmo/feature_selection/voronoi_fps.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+from time import time
+from .simple_fps import FPS
+
+VORONOI_TRIALS = 4
+
+
+class VoronoiFPS(FPS):
+    """
+    Base Class defined for Voronoi FPS methods. 
+    The method selects columns of a feature matrix X based on how dissimilar they are
+    from each other. At each point there is a set Xsel of columns, and each of the 
+    selected columns is the center of a Voronoi polyhedron. 
+    The method keeps track of which of the X columns falls within the Voronoi cell
+    of the columns of Xsel, and uses this information, as well as the distances between
+    the Xsel, to reduce the number of distances that must be computed.
```

Comment:
It would be worthwhile to re-do the docstring to better outline the algorithm. I would put something like:

```
In FPS, points are selected based upon their Hausdorff distance to 
previous selections, i.e. the minimum distance between a given point and 
any previously selected points. This implicitly constructs a Voronoi 
tessellation which is updated with each new selection, as each unselected 
point "belongs" to the Voronoi polyhedron of the nearest previous selection.

This implicit tessellation enabled a more efficient evaluation of the FPS -- 
at each iteration, we need only consider for selection those points at the 
boundaries of the Voronoi polyhedra, and when updating the tessellation 
we need only consider moving those points whose Hausdorff distance is 
greater than half of the distance between the corresponding Voronoi center 
and the newly selected point, per the triangle equality.
```

Reply:
go ahead

Reply:
Agree, edit this comment